### ISSUE TITLE
ArchiveFiles (2.211.0) Task throws Error: Archive creation failed for archive file: 

### ISSUE THREAD

**Issue by bm-fez** [microsoft/azure-pipelines-tasks#17038]
## Required Information
**Question, Bug, or Feature?**  Bug

**Enter Task Name**: ArchiveFiles (2.211.0)

## Environment
- Agent - `ubuntu-latest` Hosted 

## Issue Description
I have pipeline that builds a large Hugo Static Website. As part of this pipeline the ArchiveFiles task is used to ZIP up the pages.

This has been working without any issues, but when used today it failed with the error

`##[error]Error: Archive creation failed for archive file: ...`

I reverted the version of the ArchiveFiles task from 2.211.0 to 2.206.0 (the version that worked previously) and the pipeline completed without any errors.

Looks like there is an issue with the newer version

### Task logs

```
...
  adding: sspencer/2011/01/06/creating-your-own-identity-provider-for-windows-azure-appfabric-access-control/ 
stderr:  
error: Error: spawnSync /usr/bin/zip ENOBUFS
    at Object.spawnSync (node:internal/child_process:1111:20)
    at Object.spawnSync (node:child_process:814:24)
    at ToolRunner.execSync (/home/vsts/work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/node_modules/azure-pipelines-task-lib/toolrunner.js:879:23)
    at zipArchive (/home/vsts/work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:152:33)
    at createArchive (/home/vsts/work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:278:13)
    at doWork (/home/vsts/work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:338:9)
    at Object.<anonymous> (/home/vsts/work/_tasks/ArchiveFiles_d8b84976-e99a-4b86-b885-4849694435b0/2.211.0/archivefiles.js:347:1)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32) {
  errno: -105,
  code: 'ENOBUFS',
  syscall: 'spawnSync /usr/bin/zip',
  path: '/usr/bin/zip',
  spawnargs: [Array]
};
##[error]Error: Archive creation failed for archive file: /home/vsts/work/1/a/137827.zip 
code: 0 
stdout:   adding: .gitignore (stored 0%)
  adding: 404.html (deflated 60%)
...
```

**Comment by jabraham0301** [comment]
@bm-fez Hello, I'm facing the same issue, how can I revert the version of ArchiveFiles?, Also, is possible to resolve this issue with another alternative for `ubuntu-latest` runs?

**Comment by rfennell** [comment]
@jabraham0301 in your yaml just specify the complete version number e.g

> - task: ArchiveFiles@2.206.0

I only found this was an option recently. I previously thought you only specify the major number e.g @2, but that is not the case, full versions can be used

**Comment by mmaitre314** [comment]
+1 on this issue -- our builds just started failing on that task

**Comment by johnwc** [comment]
We are getting this same issue for private hosted agents as well as azure hosted agents.
Created an issue over here as well, in case it is an issue with nodejs in the agent. https://github.com/microsoft/azure-pipelines-agent/issues/3991

**Comment by tgolla** [comment]
Also experiencing this issue with an Azure pipeline that has worked for over a year.
##[error]Error: Archive creation failed for archive file: /home/vsts/work/1/a/2020.zip

**Comment by adovy** [comment]
Extract files task fails as well with ENOBUFS

**Comment by liester** [comment]
For what it's worth, I was able to replace the Extract Files Task with a Bash script.  
It is important to note that we use `ubuntu-18`as our VM image.  unzip is readily available.  Your scenario may require different steps.

Hopefully, it will save someone some pain.  I tried to use the `Unzip` task but it errored with some insanity.
![image](https://user-images.githubusercontent.com/6137420/195926120-db72bc25-bc43-4397-a00b-ff5fb41706a6.png)

**Comment by johnwc** [comment]
> For what it's worth, I was able to replace the Extract Files Task with a Bash script. It is important to note that we use `ubuntu-18`as our VM image. unzip is readily available. Your scenario may require different steps.
> 
> Hopefully, it will save someone some pain. I tried to use the `Unzip` task but it errored with some insanity. 

There is already a [temporary workaround](https://github.com/microsoft/azure-pipelines-tasks/issues/17038#issuecomment-1277250210) that forces it to use the previous working version of the task, rather than dealing with changing task out completely.

**Comment by sdecker** [comment]
> > For what it's worth, I was able to replace the Extract Files Task with a Bash script. It is important to note that we use `ubuntu-18`as our VM image. unzip is readily available. Your scenario may require different steps.
> > Hopefully, it will save someone some pain. I tried to use the `Unzip` task but it errored with some insanity.
> 
> There is already a [temporary workaround](https://github.com/microsoft/azure-pipelines-tasks/issues/17038#issuecomment-1277250210) that forces it to use the previous working version of the task, rather than dealing with changing task out completely.

Bizarrely, that only works for the ArchiveFile task. With ExtractFiles it refuses to accept anything but '@1'  
`Invalid step task reference ExtractFiles@1.211.0`
So I had to resort to calling unzip from bash as well.

**Comment by johnwc** [comment]
@sdecker that's because the version numbers are different for each task based on when they get updated. I believe the versions you can try for ExtractFiles would be 1.200.0, 1.198.0, 1.189.0. You can always go back to a previous working build output and see what version it was using, like shown below.

![image](https://user-images.githubusercontent.com/2798441/195944831-e3fa48d9-d9b1-45de-84ff-5ff7098a42a3.png)

**Comment by sdecker** [comment]
The archive files and extract files tasks were updated at the same time and
both have the same minor build number. I did actually cut and paste
directly from the pipeline log. The error message I pasted was when I tried
to use the latest version to see if maybe the old version was not
maintained. For some reason the pipeline refuses to let me use an exact
version number on extract files even when cutting and pasting the exact
version number.

**Comment by tgolla** [comment]
Has anyone had this magically correct itself? Last week with no pressing need to release a new site version we decided to give the issue a few days for MS to correct.  Yesterday I was able to run all our builds that had this issue with no problems creating the archive file.  Again, today we are having no issues.

**Comment by johnwc** [comment]
> Has anyone had this magically correct itself? Last week with no pressing need to release a new site version we decided to give the issue a few days for MS to correct.  Yesterday I was able to run all our builds that had this issue with no problems creating the archive file.  Again, today we are having no issues.

Go look at the failing jobs to see what version of the task it used, then what version it used when it worked today.

**Comment by johnwc** [comment]
It's not in the image, those tasks are dynamically downloaded from a repo in azure. Looks like someone rolled back the deployment for that task back to 2.206.0. I'm assuming the agent for the host it ran on is still 2.211.x?

**Comment by sdecker** [comment]
> It's not in the image, those tasks are dynamically downloaded from a repo in azure. Looks like someone rolled back the deployment for that task back to 2.206.0. I'm assuming the agent for the host it ran on is still 2.211.x?

I confirm that with the default ArchiveFiles@2 and ExtracFile@1 it reverts to the old versions and runs as expected. The agent for prior runs and todays are all 2.211.0

**Comment by LukeLangefelsTR** [comment]
I also just started running into this issue. For me the solution was changing from `ubuntu-latest` to `windows-latest`.

**Comment by DenisRumyantsev** [comment]
Fixed in `2.211.1`
